ProQuest.js

```diff
@@ -120,79 +120,96 @@ function fetchEmbeddedPdf(url, item, callback) {
-		var title = results[i].getElementsByClassName('resultTitle')[0]
-			|| results[i].getElementsByClassName('previewTitle')[0];
-		if (!title || title.nodeName != 'A') continue;
-		
-		if (checkOnly) return true;
-		found = true;
-		
-		items[title.href] = title.textContent;
+		if (results[i].offsetHeight>0) { //restrict to the active results only
+			var title = results[i].getElementsByClassName('resultTitle')[0]
+				|| results[i].getElementsByClassName('previewTitle')[0];
+			if (!title || title.nodeName != 'A') continue;
+			
+			if (checkOnly) return true;
+			found = true;
+			
+			items[title.href] = title.textContent;
+		}
 	}
 		
 	return found ? items : false;
 }
 
+var map = {
+	"newspaperArticle" : "newspaperArticle",
+	"jnlArticle" : "journalArticle", //also magazineArticles
+	"dissertationPublished" : "thesis",
+	"reportSerial" : "report",
+	"workingPaper" : "report",
+	"book" : "book",
+	"webPage" : "webpage",
+	"blog" : "blogPost",
+	"report" : "report",
+	"audioVideo" : "videoRecording",
+	"default" : "conferencePaper" //also default = manuscript?, referenceWork, encyclopedia
+}
+
 function detectWeb(doc, url) {
 	initLang(doc, url);
-
-	followLink = false;
-
+	
+	followLink = true;
+	var active = doc.getElementsByClassName('active');
+	for (var i=0; i<active.length; i++) {
+		if ( active[i].id.indexOf("tab-AbstractRecord-null") > -1) {
+			followLink = false;
+		}
+	}
+	
 	//Check for multiple first
-	if (url.indexOf('docview') == -1 &&
-		url.indexOf('pagepdf') == -1) {
-		if (getSearchResults(doc, true))
+	if (url.indexOf('docview') == -1 && url.indexOf('pagepdf') == -1) {
+		if (getSearchResults(doc, true)) {
 			return "multiple";
+		}
 	}
-
+	
+	//if we are on Abstract/Details page,
+	//then we can read the type from the corresponding field
 	var types = getTextValue(doc, ["Source type", "Document type", "Record type"]);
 	var zoteroType = getItemType(types);
-	if(zoteroType) return zoteroType;
-
-	//hack for NYTs, which misses crucial data.
-	var db = getTextValue(doc, "Database")[0];
-	if (db && db.indexOf("The New York Times") !== -1) {
-		return "newspaperArticle";
-	}
-
-	// Fall back on journalArticle-- even if we couldn't guess the type
-	if(types.length) return "journalArticle";
-
-	if (url.indexOf("/results/") === -1) {
-		//we might be on a page with a link to the abstract/metadata
-		//e.g. pdf view
-		var abstract_link = ZU.xpath(doc, '//a[@class="formats_base_sprite format_abstract"]');
-		if (abstract_link.length == 1) {
-			//let the tranlator know that, instead of scraping this page,
-			//we need to follow the link
-			followLink = true;
-			return (url.indexOf('/dissertations/') != -1)? "thesis" : "journalArticle";
-		}
+	if (zoteroType) return zoteroType;
+
+	//search for comments like
+	//<!-- Open:block:newspaperArticle -->
+	//these are also present on the pdf sites
+	var html = doc.body.innerHTML;
+	var matches = html.match(/Open:block:\S+/g);
+	var type = matches[0].substring(11);
```

Comment:
this will throw if there's no match. do
```
var type = doc.getElementById('mainContentLeft');
type = type && type.innerHTML.match(/\bopen:block:\S+/i);
type = type && map[type[0].substring(11)];
if (type) return type;
```